src/modules/context/UserAnswersContext.tsx

```diff
@@ -0,0 +1,145 @@
+import {
+  FC,
+  ReactElement,
+  createContext,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+
+import { useLocalContext } from '@graasp/apps-query-client';
+import { PermissionLevel, PermissionLevelCompare } from '@graasp/sdk';
+
+import {
+  AppDataType,
+  UserAnswerAppData,
+  getDefaultUserAnswerAppData,
+} from '@/config/appData';
+import { hooks, mutations } from '@/config/queryClient';
+import { UserAnswer, UserAnswerStatus } from '@/interfaces/userAnswer';
+
+type UserAnswersContextType = {
+  userAnswer?: UserAnswer;
+  selectAnswer: (userAnswer: UserAnswer) => void;
+  submitAnswer: () => void;
+  deleteAnswer: (id?: UserAnswerAppData['id']) => void;
+  allAnswersAppData?: UserAnswerAppData[];
+};
+
+const defaultContextValue: UserAnswersContextType = {
+  selectAnswer: () => null,
+  submitAnswer: () => null,
+  deleteAnswer: () => null,
+};
+
+const UserAnswersContext =
+  createContext<UserAnswersContextType>(defaultContextValue);
+
+export const UserAnswersProvider: FC<{
+  children: ReactElement | ReactElement[];
+}> = ({ children }) => {
+  const { data, isSuccess } = hooks.useAppData();
+  const [userAnswerAppData, setUserAnswerAppData] =
+    useState<UserAnswerAppData>();
+  const [allAnswersAppData, setAllAnswersAppData] =
+    useState<UserAnswerAppData[]>();
+  const { mutate: postAppData } = mutations.usePostAppData();
+  const { mutate: patchAppData } = mutations.usePatchAppData();
+  const { mutate: deleteAppData } = mutations.useDeleteAppData();
+  const { permission } = useLocalContext();
+
+  const isAdmin = useMemo(
+    () => PermissionLevelCompare.gte(permission, PermissionLevel.Admin),
+    [permission],
+  );
+
+  const { memberId } = useLocalContext();
+
+  useEffect(() => {
+    if (isSuccess) {
+      const allAns = data.filter(
+        (d) => d.type === AppDataType.UserAnswer,
+      ) as UserAnswerAppData[];
```

Comment:
Actually you can now ask for a specific type of appData or appSetting in the useAppData and useAppSetting hooks ! 
It also supports giving the expected data type as a generic which removes the need for such chenanigans 😄